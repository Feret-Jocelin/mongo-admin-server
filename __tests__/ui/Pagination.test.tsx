/* eslint-disable no-undef */
import React from 'react';
import { render } from '@testing-library/react';
import { NextIntlClientProvider } from 'next-intl';
import Pagination from '@/app/[locale]/components/ui/pagination/Pagination';

jest.mock('next-intl/client', () => ({
  usePathname() {
    return '';
  },
  useRouter() {
    return {
      push: () => jest.fn(),
      replace: () => jest.fn(),
    };
  },
}));
  
const locale = 'fr';
const message = require(`../../shared/locales/${locale}.json`)

const mockPagination = {
  total: 10,
  currentPage: 1,
  pageSizes: [10, 50, 100, 200],
  pageSize: 10,
}
describe("Pagination", () => {
  it('renders', () => {
    const handleChange = jest.fn();
    const handlePageSizeChange = jest.fn();
    const { container } = render(
    <NextIntlClientProvider locale={locale} messages={message}>
      <Pagination 
        total={mockPagination.total}
        currentPage={mockPagination.currentPage}
        pageSizes={mockPagination.pageSizes}
        pageSize={mockPagination.pageSize}
        onChange={handleChange}
        onPageSizeChange={handlePageSizeChange}
      />
    </NextIntlClientProvider>
    );
    expect(container).toMatchSnapshot();
  })
})