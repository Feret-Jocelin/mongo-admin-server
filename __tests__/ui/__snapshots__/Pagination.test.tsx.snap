// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders 1`] = `
<div>
  <section
    class="pagination"
  >
    <div
      class="container"
    >
      <label />
      <select
        class="select "
      >
        <option
          disabled=""
          value=""
        >
          Choisir
        </option>
        <option
          value="10"
        >
          10
        </option>
        <option
          value="50"
        >
          50
        </option>
        <option
          value="100"
        >
          100
        </option>
        <option
          value="200"
        >
          200
        </option>
      </select>
      <div
        class="iconContainer"
      >
        <div
          class="undefined "
        />
      </div>
    </div>
    <div
      class="pagination__pages"
    >
      <button
        class="button full transparent"
        disabled=""
        type="button"
      >
        <div
          class="icon "
        />
      </button>
      <ul
        class="pagination-item"
      >
        <li
          aria-current="true"
          class="pagination-item__number active"
        >
          1
        </li>
      </ul>
      <button
        class="button full transparent"
        disabled=""
        type="button"
      >
        <div
          class="icon "
        />
      </button>
    </div>
    <div
      class="pagination__jump"
    >
      Aller Ã  la page
      <div
        class="input-container"
      >
        <input
          class="input "
          type="number"
          value="1"
        />
      </div>
    </div>
  </section>
</div>
`;
